microdnf/commands/reinstall/reinstall.cpp

```diff
@@ -89,7 +89,12 @@ void CmdReinstall::run(Context & ctx) {
         result_pset |= solv_query.get_package_set();
     }
 
-    download_packages(result_pset, nullptr);
+    std::vector<libdnf::rpm::Package> download_pkgs;
+    download_pkgs.reserve(result_pset.size());
+    for (auto package : download_pkgs) {
```

Comment:
?Are you sure that you want to iterate over `download_pkgs` and not `result_pset`